freqtrade/data/converter/orderflow.py

```diff
@@ -191,8 +191,7 @@ def populate_dataframe_with_trades(
         logger.debug(f"trades.groups_keys in {time.time() - start_time} seconds")
 
         # Merge the complex data Series back into the DataFrame
-        # TODO: maybe candle_start and candle_end should be dropped before assignment?
-        dataframe["trades"] = trades_series
+        dataframe["trades"] = trades_series.drop(columns=["candle_start", "candle_end"])
```

Comment:
@TheJoeSchr Are you sure this has the effect you think it has?

i think trade_series is a series containing dicts at this point (the to_dict is where trade_series is assigned) - so this drop would probably need to happen inline to the "to_dict" conversion ...

Reply:
Good catch. I fixed it and added a test